### DISCUSSION TITLE
Can i test it into my localhost ubuntu enviorment ?

### DISCUSSION THREAD

**Discussion by mackshakya** [eusonlito/GPS-Tracker#55] [Q&A]
Hello,

I have successfully configure GPS-Tracker into my local enviorment And i'm trying to test it on my local enviorment . Also i have one more question that is teltonica devices supported by GPS-Tracker into system?

**Comment by eusonlito** [comment]
Since the GPS device has to connect to the platform, it is necessary that the environment where it is installed is directly accessible from the Internet through a port and an IP.

Yes, Teltonika is available via TCP protocol.

**Comment by mackshakya** [comment]
Can you please confirm is all info are fine for teltonica device? 
![Screenshot from 2024-04-04 15-22-16](https://github.com/eusonlito/GPS-Tracker/assets/33120443/5c0b6719-90aa-4836-84a0-2b3b5f863d4d)

also my local environment 127.0.023: 

and created protocal for server is : 
![Screenshot from 2024-04-04 15-24-18](https://github.com/eusonlito/GPS-Tracker/assets/33120443/718385fd-3428-4bda-aac2-00457dc1e02a)

**Comment by eusonlito** [comment]
Yes, you have to configure now your Teltonika device to your public IP at port 2025.

**Reply by mackshakya** [reply]
can i send data(messages) from teltonika device to 127.0.0.23:2025 as you mentioned into last reply?

**Reply by eusonlito** [reply]
f you want your device to communicate with your computer over the internet using a local IP address, you'd need to set up port forwarding on your router. This process involves configuring your router to forward all traffic coming to a specific external port (in this case, 2025) to the internal IP address and port of your local environment. Port forwarding requires a good understanding of your network's setup and how to configure your router.

In summary, while it's technically possible to connect your Teltonika device to your local computer for testing, it involves complex networking configurations and poses security risks. You'll need to have a good grasp of network settings, port forwarding, and security measures to proceed safely. 

Given the complexities and potential security risks of setting up a direct connection to your local computer, the best option would be to hire a cheap VPS for a secure and publicly accessible setup.
